### PR TITLE
Build DecisionReviewCompleted Plain Old Ruby Object Transformer

### PR DESCRIPTION
**Pull request by gcd253**
Resolves [APPEALS-61140](https://jira.devops.va.gov/browse/APPEALS-61140)

# Description
Added class called `Transformers::DecisionReviewCompleted` used to create a plain old ruby object for `DecisionReviewCompletedEvent`

Added class called `DecisionReviewIssueCompleted` used to create a plain old ruby object for the contents of `Transformers::DecisionReviewCompleted`'s `decision_review_issue_completed` array

Added class called `CompletedDecision` used to create a plain old ruby object for the contents of `DecisionReviewIssueCompleted`'s `decision` field

Added `decision_review_completed_context` Rspec context to mock `DecisionReviewCompletedEvent` payload contents for all future `DecisionReviewCompleted` testing

## Acceptance Criteria
- [x] The `DecisionReviewCompleted` active model will consist of a class called `Transformers::DecisionReviewCompleted` with the following attributes:
- `claim_id`, Integer
- `decision_review_type`, String or NilClass
- `veteran_participant_id`, String or NilClass
- `claimant_participant_id`, String or NilClass
- `remand_created`, Boolean or NilClass
- `ep_code_category`, String or NilClass
- `claim_lifecycle_status`, String or NilClass
- `actor_username`, String or NilClass
- `actor_application`, String or NilClass
- `completion_time`, String or NilClass
- `decision_review_issues_created`, Array

- [x] A class called `DecisionReviewIssueCompleted` with the following attributes:
- `decision_review_issue_id`, Integer
- `contention_id`, Integer or NilClass
- `remand_claim_id`, String or NilClass
- `remand_contention_id`, String or NilClass
- `unidentified`, boolean or NilClass
- `prior_rating_decision_id`, Integer or NilClass
- `prior_non_rating_decision_id`, Integer or NilClass
- `prior_caseflow_decision_issue_id`, Integer or NilClass
- `prior_decision_rating_sn`, String or NilClass
- `prior_decision_text`, String or NilClass
- `prior_decision_type`, String or NilClass
- `prior_decision_source`, String or NilClass
- `prior_decision_notification_date`, date or NilClass
- `legacy_appeal_issue_id`, Integer or NilClass
- `prior_decision_rating_profile_date`, string or NilClass
- `soc_opt_in`, boolean or NilClass
- `legacy_appeal_id`, Integer or NilClass
- `decision`, Hash or NilClass

- [x] A class called `CompletedDecision` with the following attributes:
- `contention_id`, Integer
- `disposition`, String
- `dta_error_explanation`, String or NilClass
- `decision_source`, String or NilClass
- `category`, String or NilClass
- `decision_id`, Integer or NilClass
- `decision_text`, String or NilClass
- `award_event_id`, Integer or NilClass
- `rating_profile_date`, String or NilClass
- `decision_recorded_time`, String or NilClass
- `decision_finalized_time`, String or NilClass

- [x] All classes will include a module that validates the class's attribute names and data types when attempting to create a `DecisionReviewCompleted` instance

- [x] RSPEC exists for this model and has 100% code coverage.

- [x] Jira Testing Exists.

## Testing Plan

1. [APPEALS-63117](https://jira.devops.va.gov/browse/APPEALS-63117)
2. [APPEALS-63109](https://jira.devops.va.gov/browse/APPEALS-63109)
3. [Jira Xray](https://jira.devops.va.gov/secure/XrayExecuteTest!default.jspa?testExecIssueKey=APPEALS-63117&testIssueKey=APPEALS-63109)

# Best practices
## Code Documentation Updates
- [x] Add or update code comments at the top of the class, module, and/or component.

### Test Coverage
Did you include any test coverage for your code? Check below:
- [x] RSpec
- [x] Other

### Code Climate
Your code does not add any new code climate offenses? If so why?
- [x] No new code climate issues added

### CODE REVIEW

File: app/models/virtual/transformers/decision_review_completed.rb [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+# CompletedDecision represents an individual decision object from
+# a decision_review_issues_completed's decision field
+class CompletedDecision
+  include MessagePayloadValidator
+
+  DECISION_ATTRIBUTES = {
+    "contention_id" => Integer,
+    "disposition" => String,
+    "dta_error_explanation" => String,
```
[the remark this answers is not in the document]
Comment (Jruuuu) [reply app/models/virtual/transformers/decision_review_completed.rb:136]: https://github.com/department-of-veterans-affairs/bip-cest-api/blob/development/bip-cest-api/src/main/resources/avros/models/Decision.avsc

**Review by chrislazo22 — APPROVED** [review]
Looks good

File: spec/models/virtual/transformers/decision_review_completed_spec.rb
Comment (Jruuuu) [inline spec/models/virtual/transformers/decision_review_completed_spec.rb:1]: Might need to test not_nil attributes with the updated transformer

**Review by Jruuuu — COMMENTED** [review]
LGTM

**Review by mattray-gov — APPROVED** [review]
LGTM

**Pull request merged**